test/Tests/src/Data/Time/Duration_Spec.enso

```diff
@@ -64,6 +65,19 @@ spec =
             duration_1>duration_2 . should_be_true
             duration_1<duration_2 . should_be_false
 
+        Test.specify "should not mix Duration and Period" <|
+            handler err = err.is_error . should_be_true
+            durations = [(Duration.new hours=1), (Duration.zero), (Duration.new hours=1 seconds=30)]
+            periods = [(Period.new days=1), (Period.new 0), (Period.new years=30), (Period.new years=3 months=2)]
+            durations.each duration->
+                periods.each period->
+                    Panic.catch Time_Error_Data (duration + period) handler
```

Comment:
Just to confirm these should be `Error` not `Panic`.

Reply:
Yes, these are `Error`, not `Panic`. I will refactor `Panic.catch` to `Error.catch`.

Reply:
Please use `should_fail_with` instead.
Alternatively use `x.catch . should_equal (the error ...)`.
We need to ensure that the tests will actually fail if the error is not thrown.